reconciletags/reconciletags.py

```diff
@@ -24,87 +24,92 @@
 import logging
 import os
 import subprocess
+import httplib2
 
 
+from containerregistry.client.v2_2 import docker_image
+from containerregistry.client.v2_2 import docker_image_list
+from containerregistry.client.v2_2 import docker_session
+from containerregistry.client import docker_creds
+from containerregistry.client import docker_name
+from containerregistry.transport import transport_pool
+
 class TagReconciler:
-    def call(self, command, dry_run, fmt="json"):
-        command += " --format=" + fmt
+    def add_tags(self, digest, tag, dry_run): 
         if not dry_run:
-            logging.debug('Running {0}'.format(command))
-            output = subprocess.check_output([command], shell=True)
-            logging.debug(output)
-            return output
+            src_name = docker_name.Digest(digest)
+            dest_name = docker_name.Tag(tag)
+            creds = docker_creds.DefaultKeychain.Resolve(src_name)
```

Comment:
This works in general, but I think in some cases we might need a different set of creds for src and dst. You should probably make another creds object below right before the push, based on the dest_name.